core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -618,6 +618,102 @@ public static ByteString substring(ByteString c, int s, int l) {
     return c.substring(s0, e0);
   }
 
+  /** SQL FORMAT_NUMBER(value, decimalOrFormat) function. */
+  public static String formatNumber(byte value, int decimalVal) {
+    DecimalFormat numberFormat = getNumberFormat(decimalVal);
+    return numberFormat.format(value);
+  }
+
+  public static String formatNumber(short value, int decimalVal) {
+    DecimalFormat numberFormat = getNumberFormat(decimalVal);
+    return numberFormat.format(value);
+  }
+
+  public static String formatNumber(int value, int decimalVal) {
+    DecimalFormat numberFormat = getNumberFormat(decimalVal);
+    return numberFormat.format(value);
+  }
+
+  public static String formatNumber(long value, int decimalVal) {
```

Comment:
do you need a method for all of these types? There's other functions that use implicit type conversion (see `COSH` or one of the trig functions for a good example). If they're necessary that's fine, just make sure they're all tested.

Reply:
Thanks for your advice, I agree with you because it does not affect the correctness of the result, and it can make code more concise.